lib/ansible/modules/cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -590,10 +599,13 @@ def ensure_route_table_present(connection, module):
         changed = changed or result['changed']
 
     if not tags_valid and tags is not None:
-        result = ensure_tags(connection, route_table.id, tags,
-                             add_only=True, check_mode=module.check_mode)
-        route_table.tags = result['tags']
-        changed = changed or result['changed']
+        if module.check_mode and not route_table:
```

Comment:
Though, I think the only time route table will be none is running in check mode. Otherwise something very wrong has happened. I can put a check in ensure_present() to fail if the route_table is none and check_mode is false before calling these other operations though. (?)